src/physics/base/CutoffParams.cc

```diff
@@ -0,0 +1,45 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file CutoffParams.cc
+//---------------------------------------------------------------------------//
+#include "CutoffParams.hh"
+#include "base/CollectionBuilder.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct on both host and device.
+ */
+CutoffParams::CutoffParams(const Input& input)
+{
+    CELER_EXPECT(input.materials);
+    CELER_EXPECT(input.particles);
+    CELER_EXPECT(!input.cutoffs.empty());
+
+    HostValue host_data;
+    host_data.num_materials = input.materials->size();
+    host_data.num_particles = input.particles->size();
+    const auto cutoffs_size = host_data.num_materials * host_data.num_particles;
+
+    auto cutoffs = make_builder(&host_data.cutoffs);
+    cutoffs.reserve(cutoffs_size);
+
+    for (const auto& per_material_cutoffs : input.cutoffs)
+    {
+        for (const auto& material_cutoffs : per_material_cutoffs.cutoffs)
+        {
+            cutoffs.push_back(std::move(material_cutoffs));
+        }
```

Comment:
Minor comments:
- `move` won't do anything here because the data is const... but it won't hurt either.
- Can you add an assertion inside the outer `for` loop that `per_material_cutoffs.size() == host_data.num_materials`?
- The `insert_back` takes two iterators so you an push a bunch of elements simultaneously, so you should be able to replace the inner loop with `cutoffs.insert_back(per_material_cutoffs.cutoffs.begin(), per_material_cutoffs.cutoffs.end());`